lib/msgpax/packer.ex

```diff
@@ -244,10 +257,19 @@ defimpl Msgpax.Packer, for: Any do
   def deriving(module, opts) do
     extractor =
       if fields = opts[:fields] do
-        quote(do: Map.take(struct, unquote(fields)))
+        quote bind_quoted: [fields: fields] do
```

Comment:
I think we should take advantage of compile time, we pop `:__struct__` from `fields` before quoting and and decide what to do after:
```elixir
quote(do: Map.take(struct, unquote(fields)))
```
or since we know `module` name upfront:
```elixir
quote do
  Map.take(struct, unquote(fields))
  |> Map.put("__struct__", unquote(module))
end
```